src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+ *
+ *     http://www.ring-team.org
+ *
+ *     RING Project
+ *     Ecole Nationale Superieure de Geologie - GeoRessources
+ *     2 Rue du Doyen Marcel Roubault - TSA 70605
+ *     54518 VANDOEUVRE-LES-NANCY
+ *     FRANCE
+ */
+
+#include <geogram/basic/attributes.h>
+#include <geogram/basic/logger.h>
+
+#include <ringmesh/basic/geometry.h>
+#include <ringmesh/geomodel/core/geomodel.h>
+#include <ringmesh/geomodel/core/geomodel_api.h>
+#include <ringmesh/geomodel/core/geomodel_geological_entity.h>
+#include <ringmesh/geomodel/core/geomodel_mesh_entity.h>
+#include <ringmesh/geomodel/core/stratigraphic_column.h>
+
+#include <ringmesh/io/geomodel_builder_resqml.h>
+
+#include <ringmesh/mesh/mesh_index.h>
+
+#include <ringmesh/mesh/surface_mesh.h>
+#include <ringmesh/mesh/volume_mesh.h>
+
+#include <ringmesh/mesh/mesh_builder.h>
+
+#include <fesapi/common/EpcDocument.h>
+#include <fesapi/common/HdfProxy.h>
+#include <fesapi/resqml2/AbstractFeatureInterpretation.h>
+#include <fesapi/resqml2_0_1/ContinuousProperty.h>
+#include <fesapi/resqml2_0_1/DiscreteProperty.h>
+#include <fesapi/resqml2_0_1/FrontierFeature.h>
+#include <fesapi/resqml2_0_1/GeneticBoundaryFeature.h>
+#include <fesapi/resqml2_0_1/Horizon.h>
+#include <fesapi/resqml2_0_1/LocalDepth3dCrs.h>
+#include <fesapi/resqml2_0_1/LocalTime3dCrs.h>
+#include <fesapi/resqml2_0_1/PropertyKind.h>
+#include <fesapi/resqml2_0_1/PropertyKindMapper.h>
+#include <fesapi/resqml2_0_1/StratigraphicColumn.h>
+#include <fesapi/resqml2_0_1/StratigraphicColumnRankInterpretation.h>
+#include <fesapi/resqml2_0_1/StratigraphicUnitInterpretation.h>
+#include <fesapi/resqml2_0_1/TectonicBoundaryFeature.h>
+#include <fesapi/resqml2_0_1/TriangulatedSetRepresentation.h>
+#include <fesapi/resqml2_0_1/UnstructuredGridRepresentation.h>
+
+/*!
+ * @brief Implementation of the class to build GeoModel from input
+ * RESQML2 .epc file
+ * @author Wan-Chiu Li
+ */
+
+namespace RINGMesh
+{
+    using namespace RESQML2_0_1_NS;
+    using namespace RESQML2_NS;
+    using namespace COMMON_NS;
+    using GMGE = GeoModelGeologicalEntity< 3 >;
+
+    namespace
+    {
+        void showAllMetadata(
+            AbstractObject* obj, const std::string& prefix = "" )
+        {
+            std::cout << prefix << "Title is : " << obj->getTitle()
+                      << std::endl;
+            std::cout << prefix << "Guid is : " << obj->getUuid() << std::endl;
+            if( !obj->isPartial() )
+            {
+                for( unsigned int i = 0; i < obj->getAliasCount(); ++i )
+                {
+                    std::cout
+                        << prefix
+                        << "Alias is : " << obj->getAliasAuthorityAtIndex( i )
+                        << ":" << obj->getAliasTitleAtIndex( i ) << std::endl;
+                }
+                for( unsigned int i = 0; i < obj->getExtraMetadataCount(); ++i )
+                {
+                    std::cout << prefix << "Extrametadata is : "
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this usage of "::std::cout" by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS106)